doc/source/train/examples/intel_gaudi/llama_deepspeed.ipynb

```diff
@@ -0,0 +1,2436 @@
+    "    # prepare training arguments\n",
+    "    training_args = prepare_training_args(config)\n",
+    "\n",
+    "    # prepare datasets\n",
+    "    # here we use dataset \"tatsu-lab/alpaca\" from huggingface\n",
+    "    # and sample some part\n",
+    "    raw_datasets = datasets.DatasetDict({\"train\": datasets.load_dataset(\"tatsu-lab/alpaca\", split='train[0:4096]')})\n",
+    "    preprocess_dataset(raw_datasets)\n",
+    "\n",
+    "    # prepare tokenizer\n",
+    "    tokenizer = AutoTokenizer.from_pretrained(config[\"model\"])\n",
+    "    tokenized_datasets = preprocess_dataset_to_tokenizer(raw_datasets, tokenizer)\n",
+    "\n",
+    "    # prepare model\n",
+    "    if config[\"deepspeed\"] is not None:\n",
+    "        auto_config = AutoConfig.from_pretrained(config[\"model\"], use_cache=False, revision=\"main\", use_auth_token=None, trust_remote_code=None)\n",
+    "        model = AutoModelForCausalLM.from_pretrained(config[\"model\"], config=auto_config, **config[\"model_config\"])\n",
+    "        model.generation_config.attn_softmax_bf16 = True\n",
+    "        model.generation_config.use_flash_attention = True\n",
+    "    else:\n",
+    "        model = AutoModelForCausalLM.from_pretrained(config[\"model\"], **config[\"model_config\"])\n",
+    "\n",
+    "    peft_config = peft.LoraConfig(**config[\"lora_config\"])\n",
+    "    model.enable_input_require_grads()\n",
+    "    model = peft.get_peft_model(model, peft_config)\n",
+    "    device = training_args.device\n",
+    "    model.to(dtype=config[\"model_config\"][\"torch_dtype\"], device=device)\n",
+    "\n",
+    "    # prepare data collator\n",
+    "    data_collator = DataCollatorForLanguageModeling(tokenizer, pad_to_multiple_of=8, return_tensors=\"pt\", mlm=False)\n",
+    "\n",
+    "    gaudi_config = GaudiConfig()\n",
+    "    gaudi_config.use_fused_adam = True\n",
+    "    gaudi_config.use_fused_clip_norm = True\n",
+    "\n",
+    "    trainer = GaudiTrainer(\n",
+    "        model=model,\n",
+    "        gaudi_config=gaudi_config,\n",
+    "        args=training_args,\n",
+    "        train_dataset=tokenized_datasets[\"train\"],\n",
+    "        eval_dataset=None,\n",
+    "        tokenizer=tokenizer,\n",
+    "        data_collator=data_collator,\n",
+    "        compute_metrics=None,\n",
+    "        preprocess_logits_for_metrics=None,\n",
+    "    )\n",
+    "\n",
+    "    train_result = trainer.train()\n",
+    "    print(f\"train_result = {train_result}\")\n",
+    "    trainer.save_model()\n",
+    "\n",
+    "    return train_result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Main Training Function\n",
+    "The `train_llama` function sets up the distributed training environment using Ray and starts the training process. To enable training using HPU, we only need to make the following changes:\n",
+    "- Set the exectuion mode for training, supported execution mode are:\n",
```

Comment:
If you have tried these 3 modes, it would be a good idea to post the e2e training times here to show the performance of each mode.

Reply:
Oh actually I'm referring to the performance of different execution modes. Like which one is faster? Does "eager.compile"> "lazy" > "eager"?

Reply:
"lazy" > "eager.compile" > "eager"